jobs/config.json

```diff
@@ -2762,6 +2762,23 @@
       "sig-scheduling"
     ]
   },
+  "ci-kubernetes-e2e-gce-dind-ipv6": {
+    "args": [
+      "--cluster=",
+      "--deployment=gce-dind",
+      "--extract=ci/latest-bazel",
+      "--gcp-zone=us-central1-f",
+      "--ip-mode=ipv6",
+      "--num-nodes=2",
+      "--provider=gce",
+      "--test_args=--ginkgo.focus=Networking|Services --ginkgo.skip=IPv4|DNS|Networking-Performance|Federation|functioning\\sNodePort|preserve\\ssource\\spod|affinity\\swork|affinity\\sfor --num-nodes=2 --minStartupPods=8",
```

Comment:
I see some of the common tests are skipped (e.g. DNS). Is it that the component is not compatible with ipv6 or the test isn't written in a compatible way? Should we not skip them so folks are aware it needs to be fixed?

Reply:
Our test coverage is described here: [IPv6 CI test coverage](https://github.com/CiscoSystems/kube-v6-test#included-test-cases). The tables on that page describe tests that are included now, failing tests that are being investigated, tests that don't make sense for our setup, and wish list tests.
As described in the "Failing Tests that are Being Investigated"](https://github.com/CiscoSystems/kube-v6-test#included-test-cases), there's an Issue filed for the DNS test coverage [k/k/ issue # 62883](https://github.com/kubernetes/kubernetes/issues/62883). A couple of mitigating factors to consider:
- Very basic kube-dns functionality is tested since the Mirantis K-D-C checks on this pod coming up (so readiness/liveness probes have to be working) before declaring the cluster up.
- Issue 62883 involves the lack of handling for PTR DNS messages in kube-dns.
- Kube-DNS will soon be replaced by coreDNS. coreDNS tests the PTR functionality, and it works there. The coreDNS folks will add IPv6 test cases with coreDNS based on the GCE-DinD framework being added here.

There are 4 failing test cases involving ESIPP (e.g. "Services [It] should be able to switch session affinity for LoadBalancer service with ESIPP on [Slow] [DisabledForLargeClusters]") that just showed up on our radar. These are disabled in our test cases, but I haven't updated our test coverage github page, nor root cause the problem. I saw your name mentioned in some ESIPP-related PRs. Do you know what these are, and whether these would be appropriate for our setup? Would this work with a GCE load balancer setup?

Reply:
Thanks, good to know the coverage is documented in detail.

ESIPP means [External Source IP Preservation](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network/external-lb-source-ip-preservation.md). This feature requires support in both of kube-proxy and cloud provider implementation of service controller. While it should work with a GCE load balancer setup, I'm not sure whether load balancer test works on a DinD cluster. I saw the `ServiceLoadBalancer` test listed in the Wish List, guessing they might be suffering similar issue?